src/helper-modules/govuk-notify/govuk-notify.service.test.ts

```diff
@@ -0,0 +1,140 @@
+
+import { GovukNotifyService } from './govuk-notify.service';
+jest.mock('notifications-node-client');
+
+describe('GovukNotifyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+
+  const govUkNotifyKey = valueGenerator.string({ length: 10 });
+  const sendToEmailAddress = valueGenerator.email();
+  const templateId = valueGenerator.string({ length: 10 });
+  const errorMessage = valueGenerator.sentence();
+  const reference = valueGenerator.base64string({ length: 32 });
+  const personalisation = {
+    firstName: valueGenerator.word(),
+    surname: valueGenerator.word(),
+    supplierName: valueGenerator.word(),
+  };
+
+  const sendEmailMethodMock = jest
+    .spyOn(NotifyClient.prototype, 'sendEmail')
+    .mockImplementation(() => Promise.resolve({ status: 201, data: expectedResponse }));
+  const loggerMock = {} as PinoLogger;
+  loggerMock.error = jest.fn();
+  const service = new GovukNotifyService(loggerMock);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('sendEmail', () => {
+    const generateNotifyError = (status: number, message: string) => {
+      const response = {
+        data: {
+          status_code: status,
+          errors: [
+            {
+              error: valueGenerator.word(),
+              message,
+            },
+          ],
+        },
+      } as AxiosResponse;
+      return new AxiosError(`Request failed with status code ${status}`, status.toString(), null, null, response);
+    };
+
+    it('calls GOV.UK Notify client constructor', async () => {
+      await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(NotifyClient).toHaveBeenCalledTimes(1);
+      expect(NotifyClient).toHaveBeenCalledWith(govUkNotifyKey);
+    });
+
+    it('calls GOV.UK Notify client sendEmail function', async () => {
+      await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation, reference });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      expect(sendEmailMethodMock).toHaveBeenCalledWith(templateId, sendToEmailAddress, { personalisation, reference });
+    });
+
+    it('returns a 201 response from GOV.UK Notify', async () => {
+      const response = await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(response).toEqual({ status: 201, data: expectedResponse });
+    });
+
+    it('calls GOV.UK Notify client with the specified field `reference` and returns a 201 response', async () => {
+      const reference = valueGenerator.string({ length: 10 });
+      const response = await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation, reference });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledWith(templateId, sendToEmailAddress, { personalisation, reference });
+      expect(response).toEqual({ status: 201, data: expectedResponse });
+    });
```

Comment:
This test case looks like it might be superfluous - isn't this covered by the two test cases above it?

Reply:
Yes, tests above do test `reference` field, it's just not their main purpose. Removing test anyway.